src/numeric_index.c

```diff
@@ -598,6 +597,8 @@ struct indexIterator *NewNumericFilterIterator(RedisSearchCtx *ctx, const Numeri
     NumericUnionCtx *uc = rm_malloc(sizeof(*uc));
     uc->lastRevId = t->revisionId;
     uc->it = it;
+    uc->sp = ctx->spec;
```

Comment:
What if the spec will get deleted

Reply:
We would have crashed before. The spec is used all over the place.

Reply:
Where is it used without reopen?

Reply:
In `NewNumericFilterIterator` we use `ctx->spec` for `IndexSpec_GetFormattedKeyByName`.

Reply:
OK, I see that we drop the cursor when the index is deleted so I guess it is fine to keep the spec. But it is not fine to keep the inverted index because it is dropped when the tree splits. Thinking about it now I also believe it is not OK on text and tags because GC might clean them.

Reply:
https://github.com/RediSearch/RediSearch/blob/ba041f443ae4bbb02696a4099036e3bbd049778b/src/fork_gc.c#L780